frontend/src/views/instructor/ddahs/import.tsx

```diff
@@ -0,0 +1,319 @@
+                oldDutyDeleted = false;
+            } else if (
+                oldDuty.description === newDuty.description &&
+                oldDuty.hours !== newDuty.hours
+            ) {
+                changes.push({ oldDuty, newDuty, status: DiffType.Updated });
+                newDuties.splice(index, 1);
+                oldDutyDeleted = false;
+                dutiesChanged = true;
+            }
+        });
+
+        // If we did not find old duties to match the new one - mark it as created
+        if (oldDutyDeleted) {
+            changes.push({ oldDuty, status: DiffType.Deleted });
+            dutiesChanged = true;
+        }
+    });
+
+    // The new duties that are left were just created - add them
+    newDuties.forEach((duty) =>
+        changes.push({ newDuty: duty, status: DiffType.Created })
+    );
+
+    // Do not return changes if everything is unchanged
+    return dutiesChanged ? changes : [];
+}
+
+function dutiesDiffList(changes: DutyDiff[]) {
+    return (
+        <ul>
+            {changes.map(({ oldDuty, newDuty, status }) => {
+                switch (status) {
+                    case DiffType.Unchanged:
+                        return (
+                            <li className="duty-unchanged">
+                                {oldDuty!.description} - {oldDuty!.hours}h
+                            </li>
+                        );
+                    case DiffType.Updated:
+                        return (
+                            <li className="duty-updated">
+                                {oldDuty!.description} - {oldDuty!.hours}h{" "}
+                                <FaArrowRight />
+                                {newDuty!.hours}h
+                            </li>
+                        );
+                    case DiffType.Created:
+                        return (
+                            <li className="duty-created">
+                                {newDuty!.description} - {newDuty!.hours}h
+                            </li>
+                        );
+                    case DiffType.Deleted:
+                        return (
+                            <li className="duty-deleted">
+                                {oldDuty!.description} - {oldDuty!.hours}h
+                            </li>
+                        );
+                    default:
+                        return null;
+                }
+            })}
+        </ul>
+    );
+}
+
+function DdahsList({ ddahs }: { ddahs: Omit<Ddah, "id">[] }) {
+    return (
+        <ul>
+            {ddahs.map(({ assignment: { applicant }, duties }, i) => (
+                <li>
+                    DDAH for {applicant.first_name} {applicant.last_name} (
+                    {applicant.utorid})
+                    <ul className="instructor-ddah-import-new-ddah-duties-list">
+                        {duties.map((duty) => (
+                            <li key={`ddah-new-${i}`}>
```

Comment:
Why the elaborate key?

Reply:
I was thinking of some examples when we might want to "undo" some updates on import, which is not a feature we are actively looking to implement as I understand. So for now no particular reason, do you want me to change that to just `i`?

Reply:
Yes, please.